x-pack/plugins/cases/public/components/connectors/servicenow/servicenow_sir_case_fields.tsx

```diff
@@ -29,250 +29,243 @@ const defaultFields: Fields = {
-                        />
-                      </EuiFlexItem>
-                    </EuiFlexGroup>
-                  </>
-                </EuiFormRow>
-              </EuiFlexItem>
-            </EuiFlexGroup>
-            <EuiFlexGroup>
-              <EuiFlexItem>
-                <EuiFormRow fullWidth label={i18n.PRIORITY}>
-                  <EuiSelect
-                    fullWidth
-                    data-test-subj="prioritySelect"
-                    hasNoInitialSelection
-                    isLoading={isLoadingChoices}
-                    disabled={isLoadingChoices}
-                    options={priorityOptions}
-                    value={priority ?? undefined}
-                    onChange={(e) => onChangeCb('priority', e.target.value)}
-                  />
-                </EuiFormRow>
-              </EuiFlexItem>
-            </EuiFlexGroup>
-            <EuiFlexGroup>
-              <EuiFlexItem>
-                <EuiFormRow fullWidth label={i18n.CATEGORY}>
-                  <EuiSelect
-                    fullWidth
-                    data-test-subj="categorySelect"
-                    options={categoryOptions}
-                    value={category ?? undefined}
-                    isLoading={isLoadingChoices}
-                    disabled={isLoadingChoices}
-                    hasNoInitialSelection
-                    onChange={(e) =>
-                      onChange({ ...fields, category: e.target.value, subcategory: null })
-                    }
-                  />
-                </EuiFormRow>
-              </EuiFlexItem>
-              <EuiFlexItem>
+          <EuiFlexGroup>
+            <EuiFlexItem>
+              <EuiFormRow fullWidth label={i18n.PRIORITY}>
+                <EuiSelect
+                  fullWidth
+                  data-test-subj="prioritySelect"
+                  hasNoInitialSelection
+                  isLoading={isLoadingChoices}
+                  disabled={isLoadingChoices}
+                  options={priorityOptions}
+                  value={priority ?? undefined}
+                  onChange={(e) => onChangeCb('priority', e.target.value)}
+                />
+              </EuiFormRow>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+          <EuiFlexGroup>
+            <EuiFlexItem>
+              <EuiFormRow fullWidth label={i18n.CATEGORY}>
+                <EuiSelect
+                  fullWidth
+                  data-test-subj="categorySelect"
+                  options={categoryOptions}
+                  value={category ?? undefined}
+                  isLoading={isLoadingChoices}
+                  disabled={isLoadingChoices}
+                  hasNoInitialSelection
+                  onChange={(e) =>
+                    onChange({ ...fields, category: e.target.value, subcategory: null })
+                  }
+                />
+              </EuiFormRow>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              {subcategoryOptions?.length > 0 ? (
```

Comment:
This is the only real change: do not show this field if no subcatergory options.